projects/enhanced-dht-proposal-with-rated-list-and-hierarchical-levels.md

```diff
@@ -0,0 +1,72 @@
+# Enhanced DHT Proposal with Rated List and Hierarchical Levels
+
+Formal definition, implementation, and verification of the rated list Distributed Hash Table (DHT) for data availability sampling (DAS).
+
+## Motivation
+
+### Introduction to DAS
+Data availability sampling in Ethereum is an attempt to solve both data integrity and data availability problems for nodes with minimal storage requirements. DAS ensures that the data is available by probing for specific samples from other nodes in the network, and then employs erasure codes for reconstruction. Full Danksharding aims to further enhance this mechanism by fully integrating data availability sampling into Ethereum's sharding design, thereby allowing each shard to independently verify data availability while maintaining the overall security and efficiency of the network.
+
+### Step towards Full danksharding
+However, there are a sufficiently large number of changes that are required to accommodate full danksharding, and to efficiently achieve the throughput that is envisioned. One such improvement lies in the structure of the Distributed Hash Table currently employed in protocol.
+Existing DHT solutions like Kademlia present a bottleneck for sample dispersal [[1]](#Resources) when block proposers are the sole seeders of the DHT.  From the perspective of security, a single malicious hop in kademlia can lead to an infinite lookup. On the other hand, it practically has no sybil resistance allowing a local portion, or even a global portion, of the keyspace to be flooded. Moreover, kademlia is 20year old DHT technology leaving it ill-optimized for DHT solutions for a use case like Ethereum. Hence, to support the security and networking requirements of full danksharding a better construction of a DHT is required.
+
+## Project description
+We develop on the idea published by Dankrad in his blog titled “The Rated List”, which enhances existing DHTs based on having a “list” of all nodes and hierarchical levels, but with some rating added in for protection. There has been supporting research [[5]](#Resources) that validate simulating a social network for a DHT for stronger sybil resistance.
+
+We start by first drawing out a formal specification of the rated list that serves as an extension of the formal definition of DAS [[3]](#Resources). Then we go on to build a Proof-of-Concept while also identifying and defining the security properties essential to the framework(Security-by-Design). Then we test against the POC for basic functionality, network analysis and security analysis. Finally, we intend to comment on the viability of the solution for full danksharding based on its robustness and integrity.
```

Comment:
I would highlight once again in the last part that the goal is to build a prototype which can be used for feasibility study and further research, not a production ready implementation